### PR TITLE
Add torchelastic check when sanitizing GPUs

### PR DESCRIPTION
**Pull request by SeanNaren**
## What does this PR do?

When using torchelastic, I noticed that `nproc_per_node` is set to one per gpu device (so if there was 8 GPUs, there are 8 processes) that only see the individual GPU. This means each process sees one GPU at GPU ID 0.

This PR skips sanitising the total GPUs when using torchelastic + one visible GPU with multi-gpu setups, assuming torchelastic has created processes.

cc @ananthsub @awaelchli 

## Before submitting
- [ ] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes? (if necessary)
- [x] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [x] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: tests/models/test_gpu.py [hunk 1]
```diff
@@ -329,3 +331,25 @@ def to(self, *args, **kwargs):
...
+@mock.patch('torch.cuda.device_count', return_value=1)
+@pytest.mark.parametrize("gpus", [[0, 1, 2], 2, '0'])
+def test_torchelastic_gpu_parsing(mocked_device_count, gpus):
+    """
+    Ensure when using torchelastic and nproc_per_node is set to the default of 1 per GPU device
+    That we omit sanitizing the gpus as only one of the GPUs is visible.
+    """
+    trainer = Trainer(gpus=gpus)
+    assert trainer.accelerator_connector.parallel_device_ids == device_parser.parse_gpu_ids(gpus)
+    assert trainer.gpus == gpus
```
Comment (awaelchli) [inline tests/models/test_gpu.py:355]: just for sanity maybe assert `isinstance(cluster_environment, TorchElasticEnvironment)`? :)

**Review by awaelchli — APPROVED** [review]

**Review by carmocca — APPROVED** [review]

**Review by justusschock — APPROVED** [review]
